src/Controls/src/Core/ViewExtensions.cs

```diff
@@ -274,6 +280,40 @@ public static Task<bool> TranslateTo(this VisualElement view, double x, double y
 
 			return tcs.Task;
 		}
+		/// <summary>
+		/// Animates an elements <see cref="VisualElement.TranslationX"/> property
+		/// from their current value to the new value. This ensures that the input layout is in the same position as the visual layout.
+		/// </summary>
+		/// <param name="view">The view on which this method operates.</param>
+		/// <param name="x">The x component of the final translation vector.</param>
+		/// <param name="length">The time, in milliseconds, over which to animate the transition. The default is 250.</param>
+		/// <param name="easing">The easing function to use for the animation.</param>
+		/// <exception cref="ArgumentNullException">Thrown when <paramref name="view"/> is <see langword="null"/>.</exception>
+		public static Task<bool> TranslateXTo(this VisualElement view, double x, uint length = 250, Easing? easing = null)
+		{
+			if (view is null)
+				throw new ArgumentNullException(nameof(view));
+
+			// Note: We use the same param 'name' as 'TranslateTo' stopping previous 'TranslateTo' animations
+			return AnimateTo(view, view.TranslationX, x, nameof(TranslateTo), (v, value) => v.TranslationX = value, length, easing);
```

Comment:
With your comment of:

> We use the same param 'name' as 'TranslateTo' stopping previous 'TranslateTo' animations

I see the `ScaleYTo` does not use the `ScaleTo` value. Is there a reason that translate is different? Is the scale a bug? Not sure if any @dotnet/maui-internal-contributors know the answer to the scale?

Reply:
I can only comment from my outside perspective that 'Translate' appears to be different than Scale. There are standard animations that use ScaleXTo and ScaleYTo at different scales for buttons and other objects, particularly in games.